### PR TITLE
feat: propagate PF NO-CARRIER to VF representors (VF state follow)

### PR DESCRIPTION
**Pull request by rollandf**
When an upstream switch port goes down the PF correctly shows NO-CARRIER but the VF representor and the VF inside the pod remain UP, causing workloads to transmit silently into a dead uplink.

Add a background monitor (internal/vfstatefollower) that polls PF carrier state every second via netlink OperState. When all PFs of a RailTopology report NO-CARRIER, the active VF representors (those whose VF is inside a pod) are set admin-down, propagating the link-down event into the pod. Carrier recovery on any PF restores the representors.

- Add LinkSetDown and IsLinkNoCarrier to the NetlinkLib interface
- Use sriovnet to enumerate VFs and resolve representor names
- Add VFStateFollow *bool field to SpectrumXRailPoolConfigSpec (nil/true = enabled, &false = disabled per CR)
- Wire the monitor into the flowscontroller binary
- 15 unit tests covering carrier logic, rep state transitions, node selector filtering, and the VFStateFollow opt-out

## Summary by CodeRabbit

* **New Features**
  * Added `vfStateFollow` to `SpectrumXRailPoolConfig` (default: enabled). A new `vf-state-follower` monitor reconciles VF representor admin state based on whether all selected PFs are reporting NO-CARRIER.

* **Bug Fixes**
  * Improved idempotent behavior for admin transitions (avoids unnecessary changes) and more resilient handling of missing node/lookup edge cases.

* **Tests**
  * Added a comprehensive Ginkgo/Gomega test suite covering selector filtering, NO-CARRIER logic, idempotency, and failure scenarios.

### CODE REVIEW

**Comment by rollandf** [comment]
@greptileai review

File: internal/vfstatefollower/monitor.go [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+		return
+	}
+
+	rpcList := &v1alpha2.SpectrumXRailPoolConfigList{}
+	if err := m.Client.List(ctx, rpcList); err != nil {
+		logr.Error(err, "failed to list SpectrumXRailPoolConfig")
+		return
+	}
+
+	for i := range rpcList.Items {
```
[the remark this answers is not in the document]
Comment (rollandf) [reply internal/vfstatefollower/monitor.go:88]: Fixed

File: internal/vfstatefollower/monitor.go [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+		rpc := &rpcList.Items[i]
+		sel := labels.Set(rpc.Spec.NodeSelector).AsSelector()
+		if !sel.Matches(labels.Set(node.Labels)) {
+			continue
+		}
```
[the remark this answers is not in the document]
Comment (rollandf) [reply internal/vfstatefollower/monitor.go:93]: When the nodeSelector changes to exclude this node, the SpectrumXRailPoolConfigHostFlowsReconciler reconciles and patches the SriovNetworkNodePolicy, which the SR-IOV operator processes by deleting the VFs on this node. When the VFs are deleted, their representors are also removed from the host. There are no stranded reps to recover — the interfaces simply cease to exist

File: internal/vfstatefollower/monitor.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+				var err error
+				allDown, err = m.allPFsNoCarrier(rt.NicSelector.PfNames)
+				if err != nil {
+					logr.Error(err, "failed to check PF carrier state", "topology", rt.Name)
+					continue
+				}
+			}
+			// When disabled, allDown is false: the restore branch runs and
+			// brings up any representors that were previously set down.
+			m.reconcileTopology(ctx, rt, allDown)
```
[the remark this answers is not in the document]
Comment (rollandf) [reply internal/vfstatefollower/monitor.go:107]: The monitor is the sole actor that sets VF representors admin-down in this system. Neither the CNI, the SR-IOV operator, nor normal operator procedures set representors down independently. The scenario requires a manual admin action that is outside the operational model of this feature. Since the monitor has exclusive ownership of this state, LinkSetUp on a down rep is always a correct restoration of state this component set. No state tracking is needed.

File: api/v1alpha2/spectrumxrailpoolconfig_types.go [hunk 4]
```diff
@@ -54,6 +54,13 @@ type RailTopology struct {
 type SpectrumXRailPoolConfigSpec struct {
 	// +kubebuilder:default:=true
 	DraEnabled bool `json:"draEnabled,omitempty"`
+	// VFStateFollow enables mirroring the PF carrier state to VF representors.
+	// When all PFs of a topology report NO-CARRIER, their representors are set
+	// admin-down so pods receive a link-down notification.
+	// Defaults to true when not set.
+	// +kubebuilder:default:=true
+	// +optional
+	VFStateFollow *bool `json:"vfStateFollow,omitempty"`
```
Comment (almaslennikov) [inline api/v1alpha2/spectrumxrailpoolconfig_types.go:63]: Need to update manifests in NetOp once this is merged

File: cmd/flowscontroller/main.go [hunk 5]
```diff
@@ -197,6 +199,13 @@ func main() {
 		os.Exit(1)
 	}
 
+	vfFollower := &vfstatefollower.Monitor{
+		Client:       mgr.GetClient(),
+		NodeName:     Options.NodeName,
+		Netlink:      netlinklib.New(),
+		PollInterval: time.Second,
```
Comment (almaslennikov) [inline cmd/flowscontroller/main.go:206]: do we want to allow to configure this interval?

File: internal/vfstatefollower/monitor.go [hunk 6]
```diff
@@ -0,0 +1,190 @@
+/*
+ Copyright 2025, NVIDIA CORPORATION & AFFILIATES
```
Comment (e0ne) [inline internal/vfstatefollower/monitor.go]: nit: 2026

File: api/v1alpha2/spectrumxrailpoolconfig_types.go (hunk 4 above)
Comment (rollandf) [reply api/v1alpha2/spectrumxrailpoolconfig_types.go:63]: Ack

File: cmd/flowscontroller/main.go (hunk 5 above)
Comment (rollandf) [reply cmd/flowscontroller/main.go:206]: Could be an option but let's keep it simple for now

File: internal/vfstatefollower/monitor.go (hunk 6 above)
Comment (rollandf) [reply internal/vfstatefollower/monitor.go]: Done
